src/tribol/physics/CommonPlane.cpp

```diff
@@ -157,302 +156,301 @@ real ComputeGapRatePressure( ContactPlaneManager& cpMgr,
+        {
+          auto& cp3 = static_cast<ContactPlane3D&>(plane);
+          for (IndexT j{0}; j < numPolyVert; ++j)
+          {
+            xVert[dim*j] = cp3.m_polyX[j];
+            xVert[dim*j + 1] = cp3.m_polyY[j];
+            xVert[dim*j + 2] = cp3.m_polyZ[j];
+          }
         }
- 
-        const real nodal_force_x1 = force_x * phi1[a];
-        const real nodal_force_y1 = force_y * phi1[a];
-        const real nodal_force_z1 = force_z * phi1[a];
 
-        const real nodal_force_x2 = force_x * phi2[a];
-        const real nodal_force_y2 = force_y * phi2[a];
-        const real nodal_force_z2 = force_z * phi2[a];
+        // instantiate surface contact element struct. Note, this is done with current 
+        // configuration face coordinates (i.e. NOT on the contact plane) and overlap 
+        // coordinates ON the contact plane. The surface contact element does not need 
+        // to be used this way, but the developer should do the book-keeping.
+        SurfaceContactElem cntctElem( dim, xf1, xf2, xVert,
+                                      num_nodes_per_face, numPolyVert,
+                                      &mesh1, &mesh2, index1, index2 );
+
+        // set SurfaceContactElem face normals and overlap normal
+        RealT faceNormal1[max_dim];
+        RealT faceNormal2[max_dim];
+        RealT overlapNormal[max_dim];
+
+        mesh1.getFaceNormal( index1, faceNormal1 );
+        mesh2.getFaceNormal( index2, faceNormal2 );
+        overlapNormal[0] = plane.m_nX;
+        overlapNormal[1] = plane.m_nY;
+        if (dim == 3)
+        {
+          overlapNormal[2] = plane.m_nZ;
+        }
 
-        if (logLevel == TRIBOL_DEBUG)
+        cntctElem.faceNormal1 = faceNormal1;
+        cntctElem.faceNormal2 = faceNormal2;
+        cntctElem.overlapNormal = overlapNormal;
+
+        // create arrays to hold nodal residual weak form integral evaluations
+        RealT phi1[max_nodes_per_face];
+        RealT phi2[max_nodes_per_face];
+        initRealArray( phi1, num_nodes_per_face, 0. );
+        initRealArray( phi2, num_nodes_per_face, 0. );
+
+        ////////////////////////////////////////////////////////////////////////
+        // Integration of contact integrals: integral of shape functions over //
+        // contact overlap patch                                              //
+        ////////////////////////////////////////////////////////////////////////
+        EvalWeakFormIntegral< COMMON_PLANE, SINGLE_POINT >
+                            ( cntctElem, phi1, phi2 );
+
+        ///////////////////////////////////////////////////////////////////////
+        // Computation of full contact nodal force contributions             //
+        // (i.e. premultiplication of contact integrals by normal component, //
+        //  contact pressure, and overlap area)                              //
+        ///////////////////////////////////////////////////////////////////////
+
+        // RealT phi_sum_1 = 0.;
```

Comment:
Consider keeping per original code, but guard in TRIBOL_USE_HOST?

Reply:
See above comment.  I think compilation failure is better here.